### ISSUE TITLE
Dirac coalescent aborts for large numbers of ancestors with c close to 0

### ISSUE THREAD

**Issue by jeromekelleher** [jeromekelleher/msprime#187]
We have an underflow problem somewhere:
```
jk@holly$ ./main simulate dev.cfg
gsl: exp.c:113: ERROR: underflow
Default GSL error handler invoked.
Aborted
```

The parameters below should reproduce the problem:
```
model = { 
     name = "dirac",
     psi = 0.99;
     c = 0.1;
};
num_loci = 1000;
recombination_map = (
    [0.0, 0.1],
    [100000.0, 0.0]
);
```

**Comment by jeromekelleher** [comment]
@shajoezhu, would you mind looking into this please?

**Comment by jeromekelleher** [comment]
Fixed in #188.
